components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -222,13 +225,84 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
     #[inline]
     pub fn apply_delete_range(
         &mut self,
-        _cf: &str,
-        _index: u64,
-        _start_key: &[u8],
-        _end_key: &[u8],
-        _notify_only: bool,
+        mut cf: &str,
+        index: u64,
+        start_key: &[u8],
+        end_key: &[u8],
+        notify_only: bool,
+        use_delete_range: bool,
     ) -> Result<()> {
-        // TODO: reuse the same delete as split/merge.
+        PEER_WRITE_CMD_COUNTER.delete_range.inc();
+        let off = data_cf_offset(cf);
+        if self.should_skip(off, index) {
+            return Ok(());
+        }
+        if !end_key.is_empty() && start_key >= end_key {
+            return Err(box_err!(
+                "invalid delete range command, start_key: {:?}, end_key: {:?}",
+                start_key,
+                end_key
+            ));
+        }
+        // region key range has no data prefix, so we must use origin key to check.
+        util::check_key_in_region(start_key, self.region())?;
```

Comment:
In multi-rocksdb, I think there's no need to check check_key_in_region as it won't impact correctness thanks to isolation.